packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_item_details/di/item_details_with_dependencies.dart

```diff
@@ -0,0 +1,39 @@
+{{> licence.dart }}
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_rx_bloc/flutter_rx_bloc.dart';
+import 'package:provider/provider.dart';
+
+import '../../base/models/item_model.dart';
+import '../blocs/item_details_bloc.dart';
+import '../views/item_details_page.dart';
+
+class ItemDetailsWithDependencies extends StatelessWidget {
+  const ItemDetailsWithDependencies({
+    required this.itemId,
+    this.item,
+    Key? key,
+  }) : super(key: key);
+
+  final String itemId;
+  final ItemModel? item;
+
+  List<RxBlocProvider> get _blocs => [
+    RxBlocProvider<ItemDetailsBlocType>(
+      create: (context) => ItemDetailsBloc(
+        context.read(),
+        itemId: itemId,
+        item: item,
+      ),
+    ),
+  ];
+
+  @override
+  Widget build(BuildContext context) => MultiProvider(
+    key: UniqueKey(),
```

Comment:
Is there a reason you do not utilise the `itemId` in building the key ?

Reply:
I'm afraid that a new bloc will be created on each hot-reload, which is not a desirable effect

Reply:
reworked